src/name/bizna/ocarmsim/OCARMSim.java

```diff
@@ -1,195 +0,0 @@
-						command_line_valid = false;
-					}
-					else screen_tier = q;
-				}
-			}
-			else if(arg.equals("-sram-size")) {
-				if(i >= args.length) {
-					System.err.println("No parameter given to -sram-size");
-					command_line_valid = false;
-				}
-				else {
-					int q = Integer.parseInt(args[i++]);
-					if(q < 0 || q > 1073741824) {
-						System.err.println("Out-of-range parameter for -sram-size");
-						command_line_valid = false;
-					}
-					else sram_quantity = q;
-				}
-			}
-			else if(arg.equals("-fsro") || arg.equals("-fsrw")) {
-				if(i >= args.length) {
-					System.err.println("No parameter given to -fs*");
-					command_line_valid = false;
-				}
-				else {
-					File basepath = new File(args[i++]);
-					if(!basepath.exists() || !basepath.isDirectory()) {
-						System.err.println("Argument to "+arg+" must be a directory, and must already exist.");
-						command_line_valid = false;
-					}
-					else
-						filesystems.add(new FSSpec(basepath, arg.equals("-fsrw")));
-				}
-			}
-			else if(arg.equals("-addrinfocmd")) {
-				if(i >= args.length) {
-					System.err.println("No parameter given to -addrinfocmd");
-					command_line_valid = false;
-				}
-				else {
-					addrinfocmd = args[i++];
-				}
-			}
-			else {
-				System.err.println("Unknown command line argument: " + arg);
-				command_line_valid = false;
-			}
-			/* TODO: -fs* -drive* */
-		}
-		if(eeprom_image_path == null) {
-			System.err.println("No EEPROM image given.");
-			command_line_valid = false;
-		}
-		if(!command_line_valid) {
-			usage();
-			System.exit(1);
-		}
-		CPU cpu = new CPU();
-		FakeMachine machine = new FakeMachine();
-		JARMArchitecture arch = new JARMArchitecture();
-		CP3 cp3 = new CP3(cpu, machine, arch);
-		cpu.mapCoprocessor(3, cp3);
-		cpu.mapCoprocessor(7, new CP7(cpu));
-		PhysicalMemorySpace mem = cpu.getMemorySpace();
-		ByteArrayRegion[] rams = new ByteArrayRegion[2];
-		if(ram_quantity > 1048576) {
-			/* we need to use two modules, since no mapping can exceed 1GiB */
-			int upper_quantity = ram_quantity - 1048576;
-			mem.mapRegion(0x40000000, rams[1] = new ByteArrayRegion(upper_quantity * 1024));
-			arch.setModule2Size(upper_quantity * 1024);
-			ram_quantity = 1048576;
-		}
-		if(ram_quantity > 0) {
-			mem.mapRegion(0x00000000, rams[0] = new ByteArrayRegion(ram_quantity * 1024));
-			arch.setModule1Size(ram_quantity * 1024);
-		}
-		ROMRegion rom = new ROMRegion(new File(eeprom_image_path));
-		mem.mapRegion(0xC0000000, rom);
-		SRAMRegion sram = new SRAMRegion(sram_image_path == null ? null : new File(sram_image_path), sram_quantity, sram_writable);
-		machine.addNode(new SimEEPROM(rom, sram));
-		arch.setModule0Size((int)sram.getRegionSize());
-		arch.setSRAMRegion(sram);
-		mem.mapRegion(0x80000000, sram);
-		for(FSSpec spec : filesystems) {
-			machine.addNode(new SimFilesystem(spec.basepath, spec.writable));
-		}
-		new SimUI(screen_tier, machine, cpu, cp3, rom, sram, rams, addrinfocmd);
-	}
-}
```

Comment:
I would prefer it if this means of describing a machine were kept. Having separate main classes for the XML style and the command line style would work for me.

Your new XML method could be the default main class, and then people like me who want the alternate way could do `java -cp ocarmsim-whatever.jar name.bizna.ocarmsim.AltLauncher`.